src/utils/daos_hdlr.c

```diff
@@ -2731,13 +2738,13 @@ fs_copy_parse_path(struct file_dfs *file, char *path,
 		uuid_copy(*p_uuid, dattr.da_puuid);
 		uuid_copy(*c_uuid, dattr.da_cuuid);
 		if (dattr.da_rel_path == NULL) {
-			strcpy(path, "/");
+			strncpy(path, "/", path_len);
```

Comment:
You can use D_STRNCPY() now, and are you checking path is valid here?

Reply:
I don't see a definition for `D_STRNCPY`, or any usage in DAOS. A different name maybe?
The caller checks that path is valid, and passes path and path_len, which this function trusts to be valid.
For example: https://github.com/daos-stack/daos/blob/7deb689ea2381bc298adc436d87cebb295e88b73/src/utils/daos_hdlr.c#L2787-L2793
Is this "safe"?

Reply:
sorry, I meant D_STRNDUP.  I was confused about your code, it's not doing an allocation at all is it, just a copy into an already allocated buffer.  My mistake.

Reply:
No worries!